app/models/enhancer_patterns.rb

```diff
@@ -0,0 +1,28 @@
+class EnhancerPatterns
+  attr_accessor :enhanced_query
+
+  def initialize(enhanced_query, term)
+    @enhanced_query = enhanced_query
+    term_pattern_checker(term)
+  end
+
+  private
+
+  def term_pattern_checker(term)
+    term_patterns.each_pair do |type, pattern|
+      @enhanced_query[type.to_sym] = match(pattern, term) if match(pattern, term).present?
+    end
+  end
+
+  def match(pattern, term)
+    pattern.match(term).to_s.strip
+  end
+
+  # term_patterns are regex patterns to be applied to the basic search box input
+  def term_patterns
+    {
+      isbn: /(ISBN-*(1[03])* *(: ){0,1})*(([0-9Xx][- ]*){13}|([0-9Xx][- ]*){10})/,
```

Comment:
I'm not familiar with isbn and issn regex at all, but I do believe that they _can_ be detected by regex. Provided we got these from their spec or something authoritative, then this is great. I like this sort of extendable data structure.

Reply:
Yeah both of these patterns are simple enough for this initial test. The issn pattern comes from Wikipedia (and is super easy). The isbn pattern from some a regex site (and is somewhat more complex but mostly only if we care about things we generally won't).

Of note, both specs include checksums to ensure they are valid, but for our use case of "does this look like an issn/isbn" I don't think that's useful. The patterns themselves (which wikipedia talks about extensively) should be good enough for us to decide if we want to run a modified search or not.

Reply:
If we end up wanting to include the checksum portion, that feels like something we can add later - so yeah I'm not worried about that issue at the moment.

There may be other needs to tweak this depending on its implementation in the near- and medium- future, but for now I think this is fine based on what you're saying. Thanks!